cpp/arcticdb/util/storage_lock.hpp

```diff
@@ -67,7 +66,8 @@ struct StorageLockTimeout : public std::runtime_error {
 };
 
 inline uint64_t get_thread_id() {
-    return folly::getCurrentThreadID();
+    auto thread_id = std::this_thread::get_id();
+    return std::hash<std::thread::id>()(thread_id);
```

Comment:
This function is only used in logging, so can return the `std::thread::id` directly provided `fmt` can handle it